test/block-cache.test.ts

```diff
@@ -0,0 +1,60 @@
+import { PollingBlockTracker, Provider } from 'eth-block-tracker';
+import { JsonRpcEngine } from 'json-rpc-engine';
+import pify from 'pify';
+import { providerFromEngine, createBlockCacheMiddleware } from '../src';
+import createHitTrackerMiddleware from './util/createHitTrackerMiddleware';
+
+function createTestSetup() {
+  // raw data source
+  // create block tracker
+  // create higher level
+  const engine = new JsonRpcEngine();
+  const provider = providerFromEngine(engine);
+
+  const blockTracker = new PollingBlockTracker({
+    provider: provider as Provider,
```

Comment:
I may know the answer to this already, but why do we need to typecast `provider`?

Reply:
the `PollingBlockTracker` has a different provider type that the `error` doesnt match:

<img width="1930" alt="image" src="https://user-images.githubusercontent.com/364566/163205031-b2e58152-f00d-493d-85f9-643957539d94.png">

I think we want to replace this old provider interface with an EIP1193 compatible one everywhere, hopefully soon.

Reply:
Ugh... right. I've also noticed that the signature for the callback that `sendAsync` takes is incorrect, as apparently it wants to take an error object at all times. I'll make a note of this, thanks.